### PR TITLE
Organisation Listing Query

### PR DESCRIPTION
**Pull request by karthik-Jay**
Added:
- firestore mock page for testing queries
   - navigate to /firestore-mock to try

Updated:
- organisation listing index page:
   - load organisation mock data from firestore on load
   - drop down filters reference enums
- organisation model
    - new listing query
    - filtering options type for use in query
    - stringify based logging for easier debugging

existing bugs:
   - org listing page performs double render due to StrictMode. This can be removed in production.

future improvements:
- one click reset of filters
- programmatic generation of drop down options via the enums
- refactor of onChange hook for each filter to reduce duplication of code
- pagination with infinite scroll 

Demo:
querying

### CODE REVIEW

File: src/features/organisation-list/index.tsx [hunk 1]
```diff
@@ -21,100 +26,126 @@ export interface IOrganization extends IFilterOptions {
...
   },
-]
+];
 
 const OrganisationList: React.FC = () => {
+  // store organisation card data
+  const [orgList, setOrgList] = useState<Organisation[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
```
Comment (karthik-Jay) [inline src/features/organisation-list/index.tsx:37]: @c6z3h can you please assist with fixing this? The query doesn't seem to update the local `orgList` variable even though I'm using `setOrgList`. Also the `useEffect` runs twice, not sure what's triggering the refresh

Comment (c6z3h) [reply src/features/organisation-list/index.tsx:37]: It does update the orgList variable, perhaps the issue faced is why it does not render as such? That would have to do with line 141 -- rendering still uses `mockOrganisations` list, instead of `orgList`.

The issue with double render is problematic though. Happened to find the solution here, Ctrl+F for **React.Strict mode is on**:
* https://stackoverflow.com/questions/60618844/react-hooks-useeffect-is-called-twice-even-if-an-empty-array-is-used-as-an-ar
TLDR; fix is to remove <React.StrictMode> in `src/index.tsx` line `15` and `20`

Besides this, this reminded me of a very useful / relevant video that visually explains how the event loop in JS works:
* https://www.youtube.com/watch?v=8aGhZQkoFbQ

Comment (karthik-Jay) [reply src/features/organisation-list/index.tsx:37]: Oh I see, so the `orgList` variable shows up with data in the logs for you? I log the orglist after the update in line 127 but it shows up blank. 
image

Yes, I have yet to update the card components so that it uses the data retrieved from firestore.

Thanks for the links, will review to try to resolve the double render.

**Comment by karthik-Jay** [comment]
Closing pull request to rename branch and rebase with master
